coffee-module/coffee-module-mp/coffee-module-mp-opentracing/src/main/java/hu/icellmobilsoft/coffee/module/mp/opentracing/extension/OpenTraceHandler.java

```diff
@@ -53,14 +56,65 @@ public OpenTraceHandler() {
     }
 
     @Override
-    public <T> T runWithTrace(Supplier<T> function, Traced traced, String operation) {
+    public <T> T runWithTraceNoException(Supplier<T> function, Traced traced, String operation) {
         Tracer tracer = openTraceResolver.resolveTracer();
         SpanBuilder spanBuilder = OpenTraceUtil.createSpanBuilder(tracer, traced, operation);
         Span span = spanBuilder.start();
         Scope scope = tracer.activateSpan(span);
         try {
             return function.get();
-        } catch (Exception e) {
+        } catch (RuntimeException e) {
```

Comment:
A checked hibarol igy nem keletkezik majd exception tag, lentebbi catch-ekre is igaz.

Reply:
Néztem én is egy darabig, hogy miért nincs fordítási hiba a `throw e;`-re, miközben az eredeti `runWithTrace()` nem deklarál checked Exception-t. A `Supplier.get()` fgv nem dob checked Exception-t, ezért ebbe az ágba a kód sosem juthatott, és így nem szólt a compiler sem. A `Supplier.get()` híváson csak RTE jöhet és a többin is RTE vagy BaseException.